src/quantum/azext_quantum/operations/job.py

```diff
@@ -0,0 +1,238 @@
+        job_id = result.stdout.decode('ascii').strip()
+        return show(cmd, job_id, resource_group_name, workspace_name)
+
+    raise CLIError("Failed to submit job.")
+
+
+def _parse_blob_url(url):
+    from urllib.parse import urlparse
+    o = urlparse(url)
+
+    account_name = o.netloc.split('.')[0]
+    container = o.path.split('/')[-2]
+    blob = o.path.split('/')[-1]
+    sas_token = o.query
+
+    return {
+        "account_name": account_name,
+        "container": container,
+        "blob": blob,
+        "sas_token": sas_token
+    }
+
+
+def output(cmd, job_id, resource_group_name=None, workspace_name=None):
+    """ Returns back the results of a Q# execution """
+    import tempfile
+    import json
+    import os
+    from azure.cli.command_modules.storage._client_factory import blob_data_service_factory
+
+    path = os.path.join(tempfile.gettempdir(), job_id)
+
+    if os.path.exists(path):
+        logger.debug("Using existing blob from %s", path)
+    else:
+        logger.debug("Downloading job results blob into %s", path)
+
+        info = WorkspaceInfo(cmd, resource_group_name, workspace_name)
+        client = cf_jobs(cmd.cli_ctx, info.subscription, info.resource_group, info.name)
+        job = client.get(job_id)
+
+        if job.status != "Succeeded":
+            return f"Job status: {job.status}. Output only available if Succeeded."
+
+        args = _parse_blob_url(job.output_data_uri)
+        blob_service = blob_data_service_factory(cmd.cli_ctx, args)
+        blob_service.get_blob_to_path(args['container'], args['blob'], path)
+
+    with open(path) as json_file:
+        data = json.load(json_file)
+        return data
+
+
+def wait(cmd, job_id, resource_group_name=None, workspace_name=None, max_poll_wait_secs=5):
+    """
+    Place the CLI in a waiting state until the job finishes execution.
+    """
+    import time
+
+    def has_completed(job):
+        return job.status in ("Succeeded", "Failed", "Cancelled")
+
+    info = WorkspaceInfo(cmd, resource_group_name, workspace_name)
+    client = cf_jobs(cmd.cli_ctx, info.subscription, info.resource_group, info.name)
+
+    # TODO: LROPoller...
+    w = False
+    poll_wait = 0.2
+    job = client.get(job_id)
+
+    while not has_completed(job):
+        print('.', end='', flush=True)
+        w = True
+        time.sleep(poll_wait)
+        job = client.get(job_id)
+        poll_wait = max_poll_wait_secs if poll_wait >= max_poll_wait_secs else poll_wait * 1.5
+
+    if w:
+        print("")
+
+    return job
+
+
+def execute(cmd, program_args, resource_group_name=None, workspace_name=None, target_id=None, project=None,
+            job_name=None, shots=None, no_build=False):
+    """
+    Submits a job for quantum execution on Azure Quantum, and waits for the result.
+    """
+    job = submit(cmd, program_args, resource_group_name, workspace_name, target_id, project, job_name, shots, no_build)
+    print("Job id:", job.id)
```

Comment:
Remove `print` here. It will break the user's stdout.

Reply:
You can use logger.

Reply:
This method is just a shortcut for `submit`. The concern here is that this is a long running method that submits a job and waits for it to finish. We would like to show the job id to the user in case he gets tired and decides to not wait and does Ctrl+C, so they can retrieve the job status/output later.
Seems like all messages in `logger` are by default not shown.
Any other mechanism I can accomplish this? Maybe use std err?

Reply:
Removed the `print`. I suppose users can always get back the job id but doing `job list`.